lib/metadataTypes/Role.js

```diff
@@ -36,29 +36,37 @@ class Role extends MetadataType {
         }, []);
         // manually add ObjectID for retrieves as it is no longer automatically returned, and is needed for updates
         fields.push('ObjectID');
-        /** @type {TYPE.SoapRequestParams} */
-        let requestParams = {
-            // filter individual roles
-            filter: {
-                leftOperand: 'IsPrivate',
-                operator: 'equals',
-                rightOperand: false,
-            },
-        };
-        if (key) {
-            // move original filter down one level into rightOperand and add key filter into leftOperand
+        let requestParams = null;
+        if (retrieveDir) {
+            // not needed during caching but certainly during retrieve
+            fields.push('PermissionSets');
```

Comment:
is there any impact of this? previously roles were retrieving fine?

Reply:
done